roles/logs_client/tasks/client.yml

```diff
@@ -47,133 +47,143 @@
-  loop: "{{ logs_class_servers }}"
-  vars:
-    ansible_python_interpreter: /usr/bin/python3
-
-- name: Sign rsyslog client certificate on rsyslog server with rsyslog CA certificate
-  ansible.builtin.command: >
-    certtool --generate-certificate
-      --load-request /tmp/{{ inventory_hostname }}.csr
-      --load-ca-certificate {{ rsyslog_remote_path_cert_dir }}/{{ rsyslog_ca_cert_file }}
-      --load-ca-privkey {{ rsyslog_remote_path_key_dir }}/{{ rsyslog_ca_key_file }}
-      --template /tmp/{{ inventory_hostname }}_client_template.csr
-      --outfile {{ rsyslog_remote_path_cert_dir }}/{{ inventory_hostname }}.pem
-  when: not remote_client_key_status.stat.exists or verify_certificate_result.rc == 1
-  changed_when: not remote_client_key_status.stat.exists or verify_certificate_result.rc == 1
-  register: client_certificates_generate_results
-  failed_when: client_certificates_generate_results.rc != 0
-  become: true
-  delegate_to: "{{ groups['jumphost'] | first }}+{{ item }}"
-  loop: "{{ logs_class_servers }}"
-  notify: restart_server_rsyslog
-  vars:
-    ansible_python_interpreter: /usr/bin/python3
-
-- name: Fetch rsyslog client certificates from rsyslog server
-  ansible.builtin.fetch:
-    src: "{{ rsyslog_remote_path_cert_dir }}/{{ inventory_hostname }}.pem"
-    dest: "{{ temp_certs_dir.path }}/"
-    flat: true
-    fail_on_missing: true
-    validate_checksum: true
-  when: not remote_client_key_status.stat.exists or verify_certificate_result.rc == 1
-  become: true
-  delegate_to: "{{ groups['jumphost'] | first }}+{{ item }}"
-  loop: "{{ logs_class_servers }}"
-  notify: restart_server_rsyslog
-  vars:
-    ansible_python_interpreter: /usr/bin/python3
-
-- name: Deploy rsyslog client certificate
-  ansible.builtin.copy:
-    src: "{{ temp_certs_dir.path }}/{{ item }}"
-    dest: "/{{ rsyslog_remote_path_cert_dir }}/"
-    owner: root
-    group: root
-    mode: '0644'
-    backup: true
-    force: true
-  when: not remote_client_key_status.stat.exists or verify_certificate_result.rc == 1
-  with_items:
-    - "{{ inventory_hostname }}.pem"
-  become: true
-  notify: client_restart_rsyslog
+- name: Configure certificates
+  when: not repo_rsyslog_ca_cert.stat.exists or not remote_client_cert_status.stat.exists or
+        ( verify_certificate_result.rc is defined and verify_certificate_result.rc != 0 ) or
+        remote_client_key_status.stat.size == 0
+  block:
+    - name: Generate private {{ rsyslog_remote_path_key_dir }}/{{ inventory_hostname }}.key on rsyslog client
+      ansible.builtin.command: >
+        certtool --generate-privkey --bits 4096
+          --outfile {{ rsyslog_remote_path_key_dir }}/{{ inventory_hostname }}.key
+      when: not repo_rsyslog_ca_cert.stat.exists or not remote_client_cert_status.stat.exists or
+        ( verify_certificate_result.rc is defined and verify_certificate_result.rc != 0 ) or
+        remote_client_key_status.stat.size == 0
```

Comment:
This is the same `when` that is now applied to the entire block, so I would expect it is no longer required here for a specific task.

Reply:
Same applies to several other tasks of this block.

Reply:
correct - it is not needed here anymore :+1: